### PR TITLE
feat: duration formatting

### PR DESCRIPTION
**Pull request by skokenes**
## Changes
Adds:
- ability to format a duration when the `# number` tag is included
- ability to list durations in a terse format using `# duration.terse`

Fixes:
- if bad duration name is supplied, it is ignored and the default duration (seconds) is used

Cleanup:
- extracts duration functions into 1 place and exports them for re-use across both old and new renderer

## Examples
Old renderer
CleanShot 2024-07-12 at 11 07 13@2x

New renderer
CleanShot 2024-07-12 at 11 07 49@2x

### CODE REVIEW

File: packages/malloy-render/src/component/render-numeric-field.ts [hunk 1]
```diff
@@ -24,6 +24,7 @@
 import {AtomicField} from '@malloydata/malloy';
 import {Currency, DurationUnit} from '../data_styles';
 import {format} from 'ssf';
+import { getText } from '../html/duration';
```
Comment (nachoarreola) [inline packages/malloy-render/src/component/render-numeric-field.ts:27]: nit: run linter?

File: packages/malloy-render/src/html/duration.ts [hunk 2]
```diff
@@ -21,16 +21,95 @@
...
 } from '../data_styles';
 import {RendererOptions} from '../renderer_types';
 import {Renderer} from '../renderer';
 import {RendererFactory} from '../renderer_factory';
+import { format } from 'ssf';
+
+export function formatTimeUnit(value: number, unit: DurationUnit, options: { numFormat?: string; terse?: boolean } = {}) {
+  let unitString = unit.toString();
+    if(options.terse) unitString = terseDurationUnitsMap.get(unit) ?? unitString;
+    else if (value === 1) {
```
Comment (nachoarreola) [inline packages/malloy-render/src/html/duration.ts:40]: nit: can we add parenthesis here? I feel it would be more readable.

**Review by nachoarreola — APPROVED** [review]

File: packages/malloy-render/src/component/render-numeric-field.ts (hunk 1 above)
Comment (skokenes) [reply packages/malloy-render/src/component/render-numeric-field.ts]: DOne

File: packages/malloy-render/src/html/duration.ts (hunk 2 above)
Comment (skokenes) [reply packages/malloy-render/src/html/duration.ts]: done

**Pull request merged**